In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.ticker as mticker
import seaborn as sns
from scipy import stats
from scipy.stats import pearsonr
from statsmodels.nonparametric.smoothers_lowess import lowess

sns.set_theme(style='whitegrid', palette='muted', font_scale=1.1)
plt.rcParams.update({'figure.dpi': 110, 'axes.spines.top': False, 'axes.spines.right': False})
fmt_k = mticker.FuncFormatter(lambda x, _: f'{int(x/1000)}k ₹' if x >= 1000 else f'{int(x)} ₹')


In [3]:
df = pd.read_csv("../DATA/Clean_Dataset.csv")

df.columns = df.columns.str.strip().str.lower()

## Formulation et test d’hypothèses


### Sur la durée du vol et le  prix

#### Hypothèses 

| | Formulation |
|---|---|
| **H₀** | Il n'existe **aucune** corrélation linéaire entre la durée et le prix → ρ = 0 |
| **H₁** | Il existe une corrélation linéaire significative entre la durée et le prix → ρ ≠ 0 |

- Test **bilatéral** (on ne présuppose pas le signe de la corrélation)
- Seuil de signification : **α = 0.05**

In [6]:
from scipy.stats import pearsonr

r, p = pearsonr(df['duration'], df['price'])

alpha = 0.05

print(f'r de Pearson : {r:.4f}')
print(f'p-value      : {p:.2e}')
print(f'r²           : {r**2:.4f}  →  la durée explique {r**2*100:.1f}% de la variance du prix')
print()
if p < alpha:
    print(f'Décision (α={alpha}) : On REJETTE H₀ — corrélation statistiquement significative.')
else:
    print(f'Décision (α={alpha}) : On NE rejette PAS H₀ — pas de corrélation détectée.')

r de Pearson : 0.2042
p-value      : 0.00e+00
r²           : 0.0417  →  la durée explique 4.2% de la variance du prix

Décision (α=0.05) : On REJETTE H₀ — corrélation statistiquement significative.


### Interprétation du test

**r ≈ +0.20** : corrélation positive **faible**. La direction est claire (plus le vol est long, plus le billet tend à coûter cher), mais l'intensité est modeste.

**p-value ≪ 0.05** : avec 300 000 observations, le test a une puissance statistique très élevée — même une corrélation très faible devient significative. Le rejet de H₀ signifie que la corrélation n'est **pas due au hasard**, pas qu'elle est forte.

**r² ≈ 4%** : la durée à elle seule n'explique que 4% de la variabilité du prix. Les 96% restants sont attribuables à d'autres facteurs (`class`, `airline`, `days_left`, etc.).

> **Nuance importante :** Significativité statistique ≠ importance pratique. Avec n=300 000, un r de 0.05 serait déjà significatif. Ici r=0.20 est significatif ET modérément utile pour la prédiction, mais insuffisant seul.